### ISSUE TITLE
Mod identifier in JSON

### ISSUE THREAD

**Issue by Laserlicht** [vcmi/vcmi#3827]
**Describe your proposal**
We should add the identifier of each mod in mod.json instead of using the folder name as identifier.

**Additional context**
The current behavior makes problems e.g. if you manually download mods from github and install it manually. Or if you rename mod folders by hand.
This makes big problems with the dependencies.

**Comment by IvanSavenko** [comment]
On the other hand this would lead to a bunch of different issues if we end up with multiple mods with same ID. Not sure which is "lesser evil" here. Simplest example - if player installs mod update manually, without removing old one first he will end up with two mods with same name.

Perhaps some combination method is better option. For example, add mod ID into json BUT use it only for installing into correct directory.

**Comment by Laserlicht** [comment]
> On the other hand this would lead to a bunch of different issues if we end up with multiple mods with same ID.

Yes of course. But this case is detectable by VCMI. E.g. Error message: "Mod xyz is installed multiple times".

**Comment by IvanSavenko** [comment]
Not just "error" message.
- how to handle when player starts vcmiclient in this state? What to load / what to ignore
- how to show this in UI in Launcher
- what to do if player clicks 'update' on this mod in Launcher

And possibly some other issues I have yet to think of, including transition from existing system

**Comment by dydzio0614** [comment]
We have more important launcher issues for now - you cannot update/delete mod if it is dependency of another mod, there is no button for mass uninstall all mods (whether VCMI extras should stay is up to discuss) etc.

And installing mod with a lot of dependencies and conflicts (like H3evo from launcher) still cannot handle resolving complex dependencies and conflicts automatically after install
